lib/instrumentation.ex

```diff
@@ -111,13 +99,29 @@ defmodule OpentelemetryAbsinthe.Instrumentation do
     :ok
   end
 
+  defp get_graphql_selections(data) do
+    case data do
+      %{blueprint: %{operations: [_ | _] = operations}} ->
+        operations
+        |> Enum.flat_map(& &1.selections)
+        |> Enum.map(& &1.name)
+        |> Enum.uniq()
+
+      _ ->
+        []
+    end
+  end
+
   # Surprisingly, that doesn't seem to by anything in the stdlib to conditionally
   # put stuff in a list / keyword list.
   # This snippet is approved by José himself:
   # https://elixirforum.com/t/creating-list-adding-elements-on-specific-conditions/6295/4?u=learts
   defp put_if(list, false, _), do: list
   defp put_if(list, true, value), do: [value | list]
 
+  defp lazy_put_if(list, false, _), do: list
```

Comment:
couldn't you just add another math to the `put_if` function with an `is_function` clause? i.e.
```elixir
defp put_if(list, true, value_fn) when is_function(value_fn), do: [value_fn.() | list]
```

Reply:
You are absolutely right, I completely forgot we could do this :bow: